Geometry_Engine/Query/PrincipledCurvatureAtParameter.cs

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Reflection;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Principled curvature is the max and min curvature of the intersection curve between the surface and a plane formed by the span of the normal vector and the principled direction.")]
+        [Input("surface", "Surface to evaluate.")]
+        [Input("u", "The parameter to evaluate at.")]
+        [Input("v", "The parameter to evaluate at.")]
+        [MultiOutput(0, "minC", "Minimum principled curvature. Curvature of the intersection of the surface and a plane spanned by the normal and minimum principle direction.")]
+        [MultiOutput(1, "maxC", "Maximum principled curvature. Curvature of the intersection of the surface and a plane spanned by the normal and maximum principle direction.")]
+        [MultiOutput(2, "minK", "Minimum principled direction. Tangent of the curve on the surface with the least curvature.")]
+        [MultiOutput(3, "maxK", "Maximum principled direction. Tangent of the curve on the surface with the most curvature.")]
+        public static Output<double, double, Vector,Vector> PrincipledCurvatureAtParameter(this NurbsSurface surface, double u, double v)
```

Comment:
```suggestion
        public static Output<double, double, Vector,Vector> PrincipalCurvatureAtParameter(this NurbsSurface surface, double u, double v)
```

File also need to be updated to reflect this.